File: orn.py
import requests
import json
from authfire import *
import matplotlib.pyplot as plt

import time
import datetime
import pyrebase

start_time = time.time() # time for execute
now = datetime.datetime.now() # now time

firebase = pyrebase.initialize_app(CONFIG)
db = firebase.database()

from_date = '2018-07-15'
to_date = now.strftime('%Y-%m-%d')
bnk_name = 'bnk48official.orn'

req = requests.get(URL + from_date + '/' + to_date + '/posts/column_chart')
toJSON = json.loads(req.content)
orn = toJSON['collectors'][bnk_name]

arr_x = [] # date 
arr_y = [] # count

for i, item in enumerate(orn):
    db.child('orn').child(item['pcid']).child('url').set('http://facebook.com/'+item['pcid'])
    db.child('orn').child(item['pcid']).child('date').set(item['date'])
    db.child('orn').child(item['pcid']).child('comment_count').set(item['info']['comments_count'])
    db.child('orn').child(item['pcid']).child('share_count').set(item['info']['shares_count'])
    db.child('orn').child(item['pcid']).child('reactions').child('like').set(item['info']['reactions']['like'])
    db.child('orn').child(item['pcid']).child('reactions').child('wow').set(item['info']['reactions']['wow'])
    db.child('orn').child(item['pcid']).child('reactions').child('sad').set(item['info']['reactions']['sad'])
    db.child('orn').child(item['pcid']).child('reactions').child('love').set(item['info']['reactions']['love'])
    db.child('orn').child(item['pcid']).child('reactions').child('haha').set(item['info']['reactions']['haha'])
    db.child('orn').child(item['pcid']).child('reactions').child('angry').set(item['info']['reactions']['angry'])
    arr_x.append(item['date'].split('-')[1]+'-'+item['date'].split('-')[2])
    arr_y.append(item['info']['reactions']['like'])

plt.title('Likes count')
plt.plot(arr_x, arr_y)
plt.xlabel('date')
plt.ylabel('count')
plt.show()

print("--- %s seconds ---" % (time.time() - start_time))

File: orn.js
var { URL } = require('./url.js')
var request = require('request')
var MongoClient = require('mongodb').MongoClient;
var urlDB = "mongodb://localhost:27017/";

getDateNow = () => {
  var today = new Date();
  var dd = today.getDate();
  var mm = today.getMonth() + 1;

  var yyyy = today.getFullYear();
  if (dd < 10) {
    dd = '0' + dd;
  }
  if (mm < 10) {
    mm = '0' + mm;
  }
  return yyyy + '-' + mm + '-' + dd
}

var fromDate = '2018-03-16'
var toDate = getDateNow()
var bnkName = 'bnk48official.orn'
var url = URL + fromDate + '/' + toDate + '/posts/column_chart'

request(url, function (error, response, body) {
  var ornToJSON = JSON.parse(body)
  var orn = ornToJSON.collectors['bnk48official.orn']
  orn.map((data) => {
    MongoClient.connect(urlDB, function (err, db) {
      if (err) throw err;
      var dbo = db.db("orn");
      var postID = data.pcid.split('_')
      var obj = [
        {
          _id: postID[1],
          url: 'http://facebook.com/' + data.pcid,
          info: {
            share_count: data.info.shares_count,
            comments_count: data.info.comments_count,
            reactions: {
              wow: data.info.reactions.wow,
              sad: data.info.reactions.sad,
              love: data.info.reactions.love,
              like: data.info.reactions.like,
              haha: data.info.reactions.haha,
              angry: data.info.reactions.angry,
            },
          },
          date: data.date
        }
      ]
      dbo.collection("ornly").insert(obj, function (err, res) {
        if (err) throw err;
        console.log('1 document insert :', postID[1])
        db.close();
      });
    });
  })
})
